### ISSUE TITLE
[java] FieldDeclarationsShouldBeAtStartOfClass: Add ability to ignore interfaces

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#345]
**Rule Set:** design.html#FieldDeclarationsShouldBeAtStartOfClass

**Code Sample demonstrating the issue:**

```java
public interface StartCaptureListener {
    void onStartCapture(boolean shouldUploadMediaToTba);
}

private static final int TAKE_PHOTO_RC = 334; // This should be fine, interfaces can be above fields
```

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
Considering there is already a flag for enums, this sounds reasonable (as maybe even for inner annotations). This should be very easy to add (almost identical to enum flag, but defaulting to false to keep backwards compatibility) shall anyone be looking for an easy issue to submit a PR.
